ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceBlobSidecarsAvailabilityChecker.java

```diff
@@ -318,22 +313,35 @@ private BlobSidecarsAndValidationResult computeFinalValidationResult(
         .getBlobSidecars()
         .forEach(blobSidecar -> validatedBlobSidecars.put(blobSidecar.getIndex(), blobSidecar));
 
-    // let's create the final list of validated BlobSidecars making sure that indices and
-    // final order are consistent
-    final List<BlobSidecar> completeValidatedBlobSidecars = new ArrayList<>();
-    validatedBlobSidecars.forEach(
-        (index, blobSidecar) -> {
-          checkState(
-              index.equals(blobSidecar.getIndex())
-                  && index.equals(UInt64.valueOf(completeValidatedBlobSidecars.size())),
-              "Inconsistency detected during blob sidecar validation");
-          completeValidatedBlobSidecars.add(blobSidecar);
-        });
-
-    if (completeValidatedBlobSidecars.size() < kzgCommitmentsSupplier.get().size()) {
-      // we haven't verified enough blobs to match the commitments present in the block
-      // this should never happen in practice. If it does, is likely a bug and should be fixed.
-      checkState(
+    final List<BlobSidecar> completeValidatedBlobSidecars =
+        new ArrayList<>(validatedBlobSidecars.values());
+
+    return BlobSidecarsAndValidationResult.validResult(
+        Collections.unmodifiableList(completeValidatedBlobSidecars));
+  }
+
+  /**
+   * make sure that final blob sidecar list is complete
+   *
+   * @param blobSidecarsAndValidationResult
+   * @return
+   */
+  private BlobSidecarsAndValidationResult performCompletenessValidation(
+      final BlobSidecarsAndValidationResult blobSidecarsAndValidationResult) {
+
+    if (blobSidecarsAndValidationResult.isFailure()
+        || blobSidecarsAndValidationResult.isNotRequired()) {
+      return blobSidecarsAndValidationResult;
+    }
+
+    try {
+      spec.atSlot(blockBlobSidecarsTracker.getSlotAndBlockRoot().getSlot())
+          .miscHelpers()
+          .verifyBlobSidecarCompleteness(
+              blobSidecarsAndValidationResult.getBlobSidecars(),
+              kzgCommitmentsFromBlockSupplier.get());
+    } catch (final IllegalArgumentException ex) {
+      checkArgument(
```

Comment:
I have some concerns on checkArgument vs checkState here, anyway we are already not super consistent with it in code. The more important is that we throw it all the way up, while we have  BlobSidecarsAndValidationResult.invalidResult(blobSidecars, throwable) which looks better in this case. Or we could at least add a comment that this could throw. Though I like .invalidResult more.